addons/website_sale_stock/models/sale_order.py

```diff
@@ -84,3 +84,14 @@ def _set_shop_warning_stock(self, desired_qty, new_qty):
             desired_qty=desired_qty, new_qty=new_qty
         )
         return self.shop_warning
+
+    def _abandoned_cart_conditions_met(self, sales):
+        super()
+        # If none of the products in the checkout are available for purchase (empty inventory, for example),
+        # then the email won't be sent.
+        for line in self.with_context(website_sale_stock_get_quantity=True):
+            product = line.product_id
+            combination_info = product._get_combination_info_variant(add_qty=line.product_uom_qty)
```

Comment:
This might need an override in `website_sale_stock_renting` as well, since it would need to check for the dates.